### PR TITLE
Patternlab/dp 30296 consolidate page header components

### PR DESCRIPTION
**Pull request by clairesunstudio**
## Areas of change:
Related tickets: [DP-30296](https://massgov.atlassian.net/browse/DP-30296), [DP-23692](https://massgov.atlassian.net/browse/DP-23692)

### 1. page-banner, page-header and illustrated header are moved out of @organisms/by-template into its own directory.
| old component path | new component path |
|--|--|
|@organisms/by-template/page-header.twig | @organisms/page-header/page-header.twig|
|@organisms/by-template/illustrated-header.twig | @organisms/illustrated-header/illustrated-header.twig|
|@organisms/by-template/page-banner.twig | @organisms/legacy-page-banner/page-banner-legacy.twig|
|| @organisms/page-banner/page-banner.twig

> These changes require reference paths to be updated on the openmass side. Adopt the new page-banner component and retire page-banner-legacy and illustrated-header.

### 2. Consolidated data fields into Page Header and Page Banner components

Page Header will be able to render all the content
Screen Shot 2023-11-08 at 4 13 13 PM

Page Banner will include Page Header, and on top of that, allows us to set theme (background color) and image decoration to Page Header. 
Screen Shot 2023-11-08 at 4 17 37 PM

1. Any content that can be rendered by Page Header will be able to render in Page Banner. 
2. Configuration options:
* Themes:
  - c-primary (Used on mass.gov for navigational pages)
  - c-primary-alt (Used on mass.gov for organizational pages)
  - c-white (Used on mass.gov for information pages)
* Image:
  - renderImageOnMobile - whether to wrap the image under the 1000px breakpoint
  - image - image path for wide screens/ 2:1 aspect ratio / max rendering size @ 800 x 400px
  - imageMedium - image path for medium screens/ 4:3 aspect ratio / max rendering size @ 600 x 450px
  - imageMobile - image path for mobile screens/ 2:1 aspect ratio / max rendering size @ 480 x 240px
 * No image: Don't pass any file to image
3. If the Page Banner doesn't have a background color other white, or an image, use Page Header instead.

> These changes will require some field names to be updated, and moved into Page Header when Page Header or Page Banner components are implemented. Refer to [page-header.md](https://github.com/massgov/mayflower/blob/74bed2a94d7288d5ae2181a56c44d845c5ad1cab/packages/patternlab/styleguide/source/_patterns/03-organisms/page-header/page-header.md)  and [page-banner.md](https://github.com/massgov/mayflower/blob/9fadadf8c338a814a879a507fdc0f8d2d2de9931/packages/patternlab/styleguide/source/_patterns/03-organisms/page-banner/page-banner.md) for data structures. In general, all content are passed to Page Header, Page Banner only handles theming and images.

#### [Page Banners](https://mayflower.digital.mass.gov/patternlab/b/patternlab/DP-30296-consolidate-page-header-components/?p=viewall-organisms-page-banner):
- [Primary with links](https://mayflower.digital.mass.gov/patternlab/b/patternlab/DP-30296-consolidate-page-header-components/?p=organisms-page-banner) - used on Service, Binder
- [Primary without image](https://mayflower.digital.mass.gov/patternlab/b/patternlab/DP-30296-consolidate-page-header-components/?p=organisms-page-banner-as-no-image) - used on Service, Binder
- [Primary Alt](https://mayflower.digital.mass.gov/patternlab/b/patternlab/DP-30296-consolidate-page-header-components/?p=organisms-page-banner-as-primary-alt) - used on Orgs
- [White](https://mayflower.digital.mass.gov/patternlab/b/patternlab/DP-30296-consolidate-page-header-components/?p=organisms-page-banner-as-white) - used on Guide

#### [Page Headers](https://mayflower.digital.mass.gov/patternlab/b/patternlab/DP-30296-consolidate-page-header-components/?p=viewall-organisms-page-header):
- [All configurations](https://mayflower.digital.mass.gov/patternlab/b/patternlab/DP-30296-consolidate-page-header-components/?p=organisms-page-header) - used on News, Info Details, Location, Rules of Court, etc.
- [Underlined](https://mayflower.digital.mass.gov/patternlab/b/patternlab/DP-30296-consolidate-page-header-components/?p=organisms-page-header-with-underline) - used on Topic

### 4. Style consolidation
Visual styles, breakpoints, and how images are rendered on mobile have changed for different usages. In general:
1. Styles and break points are consolidated. 
2. Text overlaying image option is removed - for better accessibility
3. Trapezoid style is removed from page banners - this will simplify the styles and make visible areas of the background image easier to predict at different screensizes and allow content to flow better and wrap less. 
4. Height of the page banner will be more responsive to the content height, and excess height will decrease - end users will be able see the page content quicker.
5. Image styles are consolidated - easier for authors to find images, and loading of the images will be more efficient and responsive to the rendered sizes and device widths. 
6. The background image will only be rendered on mobile for elected officials.
7. Consolidate breakpoints across different page banner styles, and set breakpoints to better match device types in reality. For example, currently the org page banner image wraps on the large desktop, pushing the content further down. 
[Mayflower Component Page Banner Inventory](https://docs.google.com/spreadsheets/d/1ZwRvnUcfc3xPA1Bs0MOczzhk1AvsB-NRJDRWw84mCsE/edit#gid=0)

> In openmass, a new image style should be created to optimize and resize the raw image. New image upload instructions should be provided All other image styles variations should be cleaned up after the conversions.

----

### Already done:
PR: https://github.com/massgov/openmass/pull/2266
- Breaking changes related to paths changes and pattern renames have been resolved
- Some minor changes and cleanup related the changes mentioned in Mayflower 

### To be done:
1. Switch page-banner-legacy and illustrated-header patterns to use page-banner and page-header. Data need to be updated according to [page-header.md](https://github.com/massgov/mayflower/blob/74bed2a94d7288d5ae2181a56c44d845c5ad1cab/packages/patternlab/styleguide/source/_patterns/03-organisms/page-header/page-header.md)  and [page-banner.md](https://github.com/massgov/mayflower/blob/9fadadf8c338a814a879a507fdc0f8d2d2de9931/packages/patternlab/styleguide/source/_patterns/03-organisms/page-banner/page-banner.md) 

[DP-30296]: https://massgov.atlassian.net/browse/DP-30296?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[DP-23692]: https://massgov.atlassian.net/browse/DP-23692?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

2. Consolidate image styles for page banners, page headers, and illustrated headers to:
  - image - image path for wide screens/ 2:1 aspect ratio / max rendering size @ 800 x 400px  (image upload minimum requirement)
  - imageMedium - image path for medium screens/ 4:3 aspect ratio / max rendering size @ 600 x 450px
  - imageMobile - image path for mobile screens/ 2:1 aspect ratio / max rendering size @ 480 x 240px (only need to resize when renderImageOnMobile is set to true)

Only set `renderImageOnMobile` to true for organization elected official pages.

### CODE REVIEW

File: packages/patternlab/styleguide/source/_patterns/03-organisms/page-banner/page-banner.twig [hunk 1]
```diff
@@ -1,121 +1,45 @@
...
-</section>
+{% set theme = pageBanner.theme ? " ma__page-banner--" ~ pageBanner.theme : " ma__page-banner--c-primary" %}
+{# If there's no image, don't let the subTitle and description text span the full width of the page #}
+{% set fullWidthText = pageBanner.image %}
+
+<div class="ma__page-banner{{theme}}">
+  <div class="ma__page-banner__container">
+    <div class="ma__page-banner__content">
+      {% set pageHeader = pageBanner.pageHeader  %}
+      {# {% set pageHeader = pageHeader|merge({"fullWidth": fullWidthText}) %} #}
```
Comment (clairesunstudio) [inline packages/patternlab/styleguide/source/_patterns/03-organisms/page-banner/page-banner.twig]: Help wanted:

I'm trying to pass the boolean value to pageHeader.fullWidth whenever the PageHeader component is used within PageBanner, but It's breaking the build with this error:
`PHP Fatal error:  Uncaught Twig\\Error\\RuntimeError: The merge filter only works with arrays or "Traversable", got "NULL" as first argument in...`

Comment (clairesunstudio) [reply packages/patternlab/styleguide/source/_patterns/03-organisms/page-banner/page-banner.twig]: To replicate, uncomment this line of code

Comment (arthurbaghdas) [reply packages/patternlab/styleguide/source/_patterns/03-organisms/page-banner/page-banner.twig]: I think this means that pageHeader is null or empty, found this snippet online, 
{% set pageHeader = pageHeader|default({})|merge({"fullWidth": fullWidthText}) %}
this is setting pageheader to be emtyp if it's not set, you can try this

Comment (clairesunstudio) [reply packages/patternlab/styleguide/source/_patterns/03-organisms/page-banner/page-banner.twig]: Thank you!! This fixes the issue!

File: packages/assets/scss/03-organisms/_page-header.scss [hunk 2]
```diff
@@ -144,26 +160,32 @@ $page-header-widget-width: 350px;
...
+    //   padding-right: $column;
+    //   width: calc(100% - #{$layout-sidebar-width});
+    // }
 
-    @media ($bp-x-large-min) {
-      padding-right: $column + ($gutter * 2);
-    }
+    // @media ($bp-x-large-min) {
+    //   padding-right: $column + ($gutter * 2);
+    // }
```
Comment (tom-fleming) [inline packages/assets/scss/03-organisms/_page-header.scss]: Can these lines be deleted instead of just commented out @clairesunstudio ?

**Review by tom-fleming — APPROVED** [review]
LGTM

**Pull request merged**